### PR TITLE
[ISSUE #2459] Fix the error in best practice doc

### LINKED ISSUE #2459: [DOC]best_practice describe sendAsyncFailed error
the https://github.com/apache/rocketmq/blob/master/docs/cn/best_practice.md said 
1.2 消息发送失败处理方式
Producer的send方法本身支持内部重试，重试逻辑如下：

至多重试2次（同步发送为2次，异步发送为0次）。

but https://github.com/apache/rocketmq/blob/734834cb560a341dd03f1c738974bc5656e04c87/docs/cn/client/java/API_Reference_DefaultMQProducer.md with retryTimesWhenSendFailed (default 2) 

and the code also show it will resend message when send failed by async mode

### PR DESCRIPTION
**Pull request by coder-zzzz**
fix #2459

### CODE REVIEW

File: docs/en/best_practice.md [hunk 1]
```diff
@@ -1,6 +1,68 @@
 #  Best practices
 
 ## 1 Producer
+### 1.1 Attention of send message
```
Comment (vongosling) [inline docs/en/best_practice.md:4]: I really appreciate your attention to help finish the english copies of the user manual. Would you like to make another pr to these parts? Let pr solve only one issue.

**Review by vongosling — CHANGES_REQUESTED** [review]

File: docs/en/best_practice.md (hunk 1 above)
Comment (coder-zzzz) [reply docs/en/best_practice.md:4]: ok， I‘ll create another issue

**Review by vongosling — APPROVED** [review]
